### PR TITLE
test(frontend): scope-correct ScorecardOverview tests (not in spec #250)

### LINKED ISSUE #315: test(frontend): scope-correct ScorecardOverview tests (not in spec #250)
Scope spillover from #250. ScorecardOverview.test.tsx added OOS during coverage ratchet — not in spec or priority list. Retained because excision breaks coverage thresholds. Automatically triaged by MarketHawk Dark Factory scope enforcement.

### PR DESCRIPTION
**Pull request by omniscient**
Closes #315

## Summary
Automated implementation for issue #315.

## Preview
⚠️ _Preview environment failed to start (preview failed: docker compose up --build failed (see preview_build.log tail below)) — endpoint validation was skipped; see the failure comment on the issue._

## Commands
```bash
# Iterate after feedback
docker compose --profile factory run --rm dark-factory "Continue issue #315"

# Tear down preview when done
docker compose --profile factory run --rm dark-factory "Close issue #315"
```

---
*Generated by MarketHawk Dark Factory*

### CODE REVIEW

File: frontend/src/pages/ScorecardOverview.test.tsx [hunk 1]
```diff
@@ -60,3 +79,70 @@ describe('ScorecardOverview', () => {
...
+    });
+  });
+
+  it("returns 90-day window for '90d'", () => {
+    expect(periodToDates('90d')).toEqual({
+      start_date: '2026-03-14',
+      end_date: '2026-06-12',
+    });
+  });
+});
```
Comment (omniscient) [inline frontend/src/pages/ScorecardOverview.test.tsx:148]: **edge-case** · **medium** — The hardcoded expected dates (e.g. `2026-06-05`, `2026-03-14`) assume the runner is in UTC: `periodToDates` derives the window from local-time `getDate()`/`setDate()` but serializes via `toISOString()` (UTC), so on a CI agent running in a non-UTC zone these `toEqual` assertions can shift by a day and flake. Pin `process.env.TZ='UTC'` (or use a UTC-normalized date helper) in the test setup to make these deterministic.

File: frontend/src/pages/ScorecardOverview.test.tsx [hunk 2]
```diff
@@ -60,3 +79,70 @@ describe('ScorecardOverview', () => {
...
+
+describe('ScorecardOverview — period → useScorecard call args', () => {
+  const FIXED_DATE = new Date('2026-06-12T00:00:00.000Z');
+
+  beforeEach(() => {
+    vi.useFakeTimers();
+    vi.setSystemTime(FIXED_DATE);
+    mockUseScorecard.mockClear();
+  });
+
```
Comment (omniscient) [inline frontend/src/pages/ScorecardOverview.test.tsx:91]: **maintainability** · **low** — The `FIXED_DATE` constant plus `vi.useFakeTimers()/setSystemTime/useRealTimers` boilerplate is copy-pasted across the three new `describe` blocks; extract a shared `beforeEach/afterEach` helper to reduce duplication and drift.

**Review by omniscient — COMMENTED** [review]
## 🏭 Dark Factory Code Review

**0** blocking · **2** advisory finding(s).
